src/pip/_internal/cli/parser.py

```diff
@@ -150,7 +151,7 @@ def check_default(self, option, key, val):
         try:
             return option.check_value(key, val)
         except optparse.OptionValueError as exc:
-            print("An error occurred during configuration: %s" % exc)
+            print("An error occurred during configuration: {}".format(exc))
```

Comment:
Geez. I'm zealously looking a f-strings now.

Reply:
Soon. 
Pip Python2.7 support is ending soon.. :last_quarter_moon:

Reply:
Pip will be supporting 2.7 for a fair while yet. I'd like it to be soon, but the reality is that we're [still discussing](https://discuss.python.org/t/packaging-and-python-2/662) how we handle the matter. It's possible the decision may end up being taken out of our hands, if enough of our vendored dependencies drop Python 2 support, but that in itself is a situation that we've not fully decided what to do about.

Sorry to be pedantic about this, but I really don't want to end up chasing rumours that "pip is dropping Python 2 support" before we're ready with a proper transition plan.

PS f-strings won't be something we can use in pip until we drop support for Python <3.6. But yes, they are absolutely awesome, and I use them every chance I get.

Reply:
I’m quite sure pip will drop Python <3.6 support much sooner Python 2 😞